File: Problems/1758. Minimum Changes To Make Alternating Binary String/1758. Minimum Changes To Make Alternating Binary String.py
class Solution:
    def minOperations(self, s: str) -> int:
        def countCost(s: str, flag: str):
            res = 0
            for bit in s:
                if bit != flag: res += 1
                flag = '0' if flag == '1' else '1'
                
            return res
        
        return min(countCost(s, '0'), countCost(s, '1'))

File: Problems/1758. Minimum Changes To Make Alternating Binary String/1758. Minimum Changes To Make Alternating Binary String.java
import java.lang.Math;
class Solution {
    public int minOperations(String s) {
        int cost = 0;
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) != '0'+(i%2)) cost++;
        }

        return Math.min(cost, s.length()-cost);
    }
}